src/app/message/message.component.ts

```diff
@@ -69,10 +83,29 @@ export class MessageComponent implements OnInit, OnDestroy {
         if (page > 0 && data.length === 0) {
           this.noMorePages = true;
         }
+
+        this.extractUsernames(data);
       }, () => this.loadingPage = false);
   }
 
   /**
+   * Extract and sort all usernames from message data
```

Comment:
Yeah, that's not possible with this function. I'll try to add a request route to solve that problem.